backend/app/api/admin_routes/retrieve.py

```diff
@@ -0,0 +1,27 @@
+from typing import Optional, List
+
+from fastapi import APIRouter
+from pydantic import BaseModel
+from app.models import Document
+from app.api.deps import SessionDep, CurrentSuperuserDep
+from app.rag.retrieve import RetrieveService
+
+router = APIRouter()
+
+
+class RetrieveDocumentRequest(BaseModel):
+    chat_engine: str = "default"
+    question: str
+    top_k: Optional[int] = None
+
+
+@router.get("/admin/retrieve/documents")
+async def retrieve_documents(
+        session: SessionDep,
+        user: CurrentSuperuserDep,
+        question: str,
+        chat_engine: str = "default",
+        top_k: Optional[int] = None) -> List[Document]:
```

Comment:
Pleas use `rye format app` to lint code.

Reply:
Ran and formatted.